cmake/compiler/clang/target_arm.cmake

```diff
@@ -25,7 +31,11 @@ if(CONFIG_FPU)
   elseif(CONFIG_FP_SOFTABI)
     list(APPEND ARM_C_FLAGS   -mfloat-abi=softfp)
   endif()
-endif()
+else()
+  if (CONFIG_ARMV8_M_MAINLINE)
```

Comment:
There is definitely something wrong with these lines:
- Why it's necessary to disable FPU? Can't we just leave the default setting?
- Why disabling it only when `CONFIG_ARMV8_M_MAINLINE` is set? How about `CONFIG_ARMV8_M_BASELINE`?
- Why `endif() # CONFIG_CORTEX_M AND CONFIG_ARMV8_1_M_MAINLINE` doesn't match
`if (CONFIG_ARMV8_M_MAINLINE)`?

Reply:
> Why it's necessary to disable FPU? Can't we just leave the default setting?

Ok, now I know why `-mfpu=none` must be passed #74314